src/editor/codemirror/dnd.ts

```diff
@@ -40,7 +41,11 @@ export type CodeInsertType =
   /**
    * A function call.
    */
-  | "call";
+  | "call"
+  /**
+   * Context is unknown, used for paste events.
+   */
+  | "unknown";
```

Comment:
Presumably we could know in a similar way to dnd? Let's discuss.

Reply:
Does the unknown case happen anymore? It looks like the data attribute takes care of it.

If we systematically set the data attribute then we could skip handling anything without it (assuming it's just other random HTML on the clipboard).

Reply:
I've just removed it. If we somehow end up with html/text on the clipboard without an insert type, it pastes just fine, as it would for an "example".

Reply:
[`4eaa87b`](https://github.com/microbit-foundation/python-editor-next/pull/766/commits/4eaa87bfc6ceea88e24cb03ebca2a89fa9cbea7c).

Reply:
When reading from the clipboard let's say there's no "code" if there's no data-attribute. We always set it and we're best off not trying to paste HTML from elsewhere.

Reply:
[`6d4d186`](https://github.com/microbit-foundation/python-editor-next/pull/766/commits/6d4d186273f2ad7499dffee1509852d87ddff711).